### DISCUSSION TITLE
Add login/oauth option in order to remove ads for subscribed channels

### DISCUSSION THREAD

**Discussion by ccrowles** [NSMY/Stream-Downloader-Util#10] [Ideas]
As per title, currently Stream Downloader Util has no way of skipping the 30-seconds ads when recording livestreams, even when you're subscribed to the channel.
It would be nice if it could be implemented, since I saw that [Streamlink supports oauth tokens](https://streamlink.github.io/latest/cli/plugins/twitch.html#authentication) (albeit you have to manually import them from the browser each time).

(Also I don't know if it's already implemented, but there is also the `--twitch-disable-ads` argument for embedded ads)

**Comment by NSMY** [comment]
Hi @ccrowles , this Script is basically a CLI that combines Streamlink and ffmpeg features into a more compact experience and was intened mainly for personal use (never thought anyone would see it) to download (Mostly VODs) streams (named poorly), this was the first script of over 10 lines ive ever made during the first week of learing python & did a big refactor 2 months ago. You are more than welcome to make Alterations yourself if you want to.

**HOWEVER WHAT YOU HAVE REQUESTED** seems acceptable/able to add, i have started to look into adding it. it will probably be an alternative option (inquirer, another Multi-Choice) after size option, also working on other changes ive wanted to make

**Comment by NSMY** [comment]
@ccrowles Check [Latest Build](https://github.com/NSMY/Stream-Downloader-Util/releases/tag/v3.2-Beta) For changes both requests should be working
